Jenkinsfile

```diff
@@ -1,55 +1,64 @@
-        PROJECT_ID = 'devops-testing-419206' // Replace with your actual project ID
-        REGION = 'asia-south1' // Replace with your actual region
-    }
-
-    triggers {
-        githubPullRequest {
-            triggerPhrase('deploy') // Phrase used in PR comments to trigger deployments
-            onlyTriggerPhrase(true) // Only trigger on the specified phrase
-        }
-    }
 
+    environment {
+        // Define environment variables if needed
     stages {
         stage('Checkout') {
             steps {
                 // Checkout the repository
-                git branch: 'main', url: 'https://github.com/your-username/your-repository.git'
+                git branch: 'main', url: 'https://github.com/syeda-farha-2023/Jenkins-project.git'
+            }
+        }
+
+        stage('Activate Service Account') {
+            steps {
+                withCredentials([file(credentialsId: 'channelpay-cloud-function-admin-sa', variable: 'SERVICE_ACCOUNT_KEY_PATH')]) {
+                    sh 'gcloud auth activate-service-account --key-file="${SERVICE_ACCOUNT_KEY_PATH}"'
+                }
             }
         }
+
         stage('Detect Changes') {
             steps {
                 script {
-                    // Get the list of modified files from the PR
-                    def changedFiles = sh(script: 'git diff --name-only origin/main...HEAD', returnStdout: true).trim().split('\n')
-                    
-                    // Initialize an empty list to store updated cloud functions
-                    def updatedFunctions = []
+                    withCredentials([file(credentialsId: 'channelpay-cloud-function-admin-sa', variable: 'SERVICE_ACCOUNT_KEY_PATH')]) {
+                        // Get the list of modified files from the PR
+                        def changedFiles = sh(script: 'git diff --name-only origin/main...HEAD', returnStdout: true).trim().split('\n')
+
+                        // Initialize an empty list to store updated cloud functions
+                        def updatedFunctions = []
 
-                    // Detect which cloud functions have been modified
-                    for (file in changedFiles) {
-                        if (file.startsWith('cloud_function1/')) {
-                            updatedFunctions.add('cloud_function1')
-                        } else if (file.startsWith('cloud_function2/')) {
-                            updatedFunctions.add('cloud_function2')
+                        // Detect which cloud functions have been modified
+                        for (file in changedFiles) {
+                            if (file.startsWith('cloud_function1/')) {
+                                updatedFunctions.add('cloud_function1')
+                            } else if (file.startsWith('cloud_function2/')) {
+                                updatedFunctions.add('cloud_function2')
+                            }
                         }
```

Comment:
This won't be static. folder can be anything, also there can be n number of functions.